code/datums/smells/smell.dm

```diff
@@ -0,0 +1,88 @@
+#define SMELL_HORRIBLE 0
+#define SMELL_FOUL 1
+#define SMELL_NEUTRAL 2
+#define SMELL_PLEASANT 3
+#define SMELL_NICE 4
+
+#define SMELL_TRACE 2
+#define SMELL_FAINT 4
+#define SMELL_NORMAL 8
+#define SMELL_INTENSE 16
+
+/datum/smell
+	var/qty
+	var/name
+	var/intensity
+	var/description
+
+/datum/smell/New(var/q)
+	qty=q
+	return src
+
+
+//outputs a string that fully describes a smell
+/mob/living/proc/eval_smell(var/datum/smell/S)
+	if(ishuman(src))
+		var/mob/living/carbon/human/H = src
+		if((H.species.flags & NO_BREATHE) || (NO_BREATHE in H.mutations) || wear_mask)//can't breathe can't smell, masks block smells
+			return
+	if(!S || constipated)//if there's no smell at all OR if your nose is constipated
+		src << "<span class='notice'>You can't smell anything!</span>"
+		return
+
+	var/int = null
+	var/class = null
+	var/qty = null
+
+	var/actual_qty =  get_species_coef(src) * S.qty
```

Comment:
This doesn't check anywhere if the smell is even actually a smell datum, it will cause a runtime error if something is passed to it that isn't a smell.